### PR TITLE
nvimpager: unbreak darwin

### PR DESCRIPTION
**Pull request by sielicki**
Breakage was caused by a call to `script -c` which is unique to util-linux.
Use a naked call instead on darwin.

### CODE REVIEW

File: pkgs/tools/misc/nvimpager/default.nix [hunk 1]
```diff
@@ -32,12 +32,16 @@ stdenv.mkDerivation rec {
   checkInputs = [ lua51Packages.busted util-linux neovim ];
```
Comment (winterqt) [inline pkgs/tools/misc/nvimpager/default.nix:32]: You need to add `script` for Darwin here too, `unixtools.script` will do that (see [here](https://github.com/NixOS/nixpkgs/blob/9c16fe5b873a2919ad324aeb75fa5aa8f5da0c2c/pkgs/top-level/unixtools.nix#L142)).

```suggestion
  checkInputs = [ lua51Packages.busted unixtools.script neovim ];
```

**Review by winterqt — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/nvimpager/default.nix (hunk 1 above)
Comment (sielicki) [reply pkgs/tools/misc/nvimpager/default.nix]: Done. Thanks.

File: pkgs/tools/misc/nvimpager/default.nix [hunk 2]
```diff
@@ -29,15 +29,19 @@ stdenv.mkDerivation rec {
...
-  checkInputs = [ lua51Packages.busted util-linux neovim ];
+  checkInputs = [ lua51Packages.busted unixtools.script neovim ];
   checkPhase = ''
     runHook preCheck
-    script -c "busted --lpath './?.lua' test"
+  '' + (
+  if !stdenv.isDarwin then ''
+    ${unixtools.script}/bin/script -c "busted --lpath './?.lua' test"
+  '' else ''
+    ${unixtools.script}/bin/script "busted --lpath './?.lua' test"
```
Comment (winterqt) [inline pkgs/tools/misc/nvimpager/default.nix:39]: `script` on Darwin requires passing a file to also pass a command:
```suggestion
    ${unixtools.script}/bin/script /dev/null busted --lpath './?.lua' test
```

However, the tests fail:
```
Failure → test/nvimpager_spec.lua @ 237
cat mode with modeline ignores mode lines in git log diffs #osx_pending
test/nvimpager_spec.lua:241: Expected objects to be equal.
Passed in:
(string) 'Error detected while processing modelines:
line   19:
E518: Unknown option: foobar=errorcommit 3d032f9e546fc8a1e987bbbef54e8dab1ceaf4bd
Author: Lucas Hoffmann <lucc@posteo.de>
Date:   Mon Aug 20 23:37:04 2018 +0200

    This is based on the log entry of 3d032f9e546fc8a1e987bbbef54e8dab1ceaf4bd
    but was modiefied for the tests.

    The last line is turned into a valid modeline even with the diff prefix.
    If it is interpreted it should result in an error message.

diff --git a/test/fixtures/help.txt b/test/fixtures/help.txt
index ad4baf6..ab64439 100644
--- a/test/fixtures/help.txt
+++ b/test/fixtures/help.txt
@@ -1,3 +1,3 @@
-vim: filetype=help
 *conceal-test* normal
 conceal-test
+ vim: foobar=error
'
Expected:
(string) 'commit 3d032f9e546fc8a1e987bbbef54e8dab1ceaf4bd
Author: Lucas Hoffmann <lucc@posteo.de>
Date:   Mon Aug 20 23:37:04 2018 +0200

    This is based on the log entry of 3d032f9e546fc8a1e987bbbef54e8dab1ceaf4bd
    but was modiefied for the tests.

    The last line is turned into a valid modeline even with the diff prefix.
    If it is interpreted it should result in an error message.

diff --git a/test/fixtures/help.txt b/test/fixtures/help.txt
index ad4baf6..ab64439 100644
--- a/test/fixtures/help.txt
+++ b/test/fixtures/help.txt
@@ -1,3 +1,3 @@
-vim: filetype=help
 *conceal-test* normal
 conceal-test
+ vim: foobar=error
'

Failure → test/nvimpager_spec.lua @ 722
parent detection detects git correctly #osx_pending
test/nvimpager_spec.lua:725: Expected objects to be equal.
Passed in:
(string) 'nil'
Expected:
(string) 'git'

Failure → test/nvimpager_spec.lua @ 728
parent detection detects man correctly #osx_pending
test/nvimpager_spec.lua:731: Expected objects to be equal.
Passed in:
(string) 'nil'
Expected:
(string) 'man'

Failure → test/nvimpager_spec.lua @ 741
parent detection handles man #osx_pending
test/nvimpager_spec.lua:745: Expected objects to be equal.
Passed in:
(string) 'TEST(0)                                                                TEST(0)

SYNOPSIS
       exit

AUTHORS
       Author.

                                     date                              TEST(0)
'
Expected:
(string) 'TEST(0)                                                                TEST(0)

SYNOPSIS
       exit

AUTHORS
       Author.

                                     date                              TEST(0)
'
```

**Review by winterqt — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/nvimpager/default.nix (hunk 2 above)
Comment (SuperSandro2000) [inline pkgs/tools/misc/nvimpager/default.nix:39]:
```suggestion
    script -c "busted --lpath './?.lua' test"
  '' else ''
    script "busted --lpath './?.lua' test"
```

**Comment by SuperSandro2000** [comment]
Result of nixpkgs-review pr 185500 run on x86_64-linux 1

<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>bat-extras.batdiff</li>
    <li>bat-extras.batgrep</li>
    <li>bat-extras.batman</li>
    <li>bat-extras.batpipe</li>
    <li>bat-extras.batwatch</li>
    <li>bat-extras.prettybat</li>
  </ul>
</details>

**Comment by sielicki** [comment]
This PR is totally wrong, in the process of submitting #199797 i rebased my nixpkgs fork and polluted this PR. Gross.

Closing the other PR and abandoning this one.

**Comment by SuperSandro2000** [comment]
Uhm..

do we need to revert something? Can you resubmit this original as a new one?

**Comment by sielicki** [comment]
it’s fine, original CR was dead for months, probably not relevant anymore.

**Pull request merged**